### DISCUSSION TITLE
Permissive license such as MIT?

### DISCUSSION THREAD

**Discussion by jzazo** [Marco-Sulla/python-frozendict#63] [General]
Hi Marco. Thanks for making this package, it looks really nice. However, I am blocked in my project from using non-permissive licenses such as LGPL. The original project had MIT License. Would you consider changing the license to a permissive one? I completely understand if you have your reasons for keeping it as is. I didn't see any discussions about it. Thanks a lot.

**Comment by Marco-Sulla** [comment]
LGPL is permissive. Indeed its name is Lesser GPL.

**Reply by jzazo** [reply]
Thanks Marco. It is still somehow limiting, and it warrants a review process from compliance team in the company I work for every time it gets updated. I will not be able to use it because of that, but thanks for your answer.

**Comment by Marco-Sulla** [comment]
What are the limits you have problems with?
Furthermore, the license remains the same at every update, so why your company should do a review each time?

**Reply by jzazo** [reply]
I will quote my manager on this: "it will get flagged by component governance and will need a human review to make sure we're using it in a compliant way. With the current tooling, each time the version changes it will get flagged for review again".

We could use it, but it would add burden to my manager to explain how we use it, and I would prefer not to do that.

Thanks for following up on this in any case.

**Comment by Marco-Sulla** [comment]
I think Javier is no more interested in it, since more than 3 years passed. Anyway, I see two upvotes to its question --that's *huge* for this project X-D -- so the problem is not so much restricted.

Does the problems with LGPLv3 will be fixed if frozendict will change its license to MPL v2?

https://www.mozilla.org/en-US/MPL/2.0/

**Reply by jzazo** [reply]
Hi Marco, I am not using the library anymore, but MPLv2 is similar to LGPLv3 in being a weak copyleft license, similar restrictions would apply.

I also think it would not be a blocker to release code depending on it, if I wanted to release a project that depended on a library with this type of license.

**Reply by Marco-Sulla** [reply]
> I also think it would not be a blocker to release code depending on it, if I wanted to release a project that depended on a library with this type of license.

Yes, of course you can publish code that depends on a library licensed under LGPL or MPL. You can also add it to closed-source code. 

The only restriction with MPL is that, if you want to create your own version of the library, you can't use a more restrictive license, e.g. a proprietary license.

The difference with LGPL is that LGPL is much more restrictive. For example, if someone want to include a library released under LGPL in statically linked fashion, LGPL imposes additional and boring limits.
